pkg/proxy/roundrobin.go

```diff
@@ -210,6 +221,10 @@ func (lb *LoadBalancerRR) OnUpdate(endpoints []api.Endpoints) {
 	for _, endpoint := range endpoints {
 		existingEndpoints, exists := lb.endpointsMap[endpoint.Name]
 		validEndpoints := filterValidEndpoints(endpoint.Endpoints)
+		// Need to compare sorted endpoints here, since they are shuffled below
+		// before being put into endpointsMap
+		sort.Strings(existingEndpoints)
```

Comment:
This should be easy to confirm. If we need to copy:
http://blog.golang.org/go-slices-usage-and-internals